### PR TITLE
feat(frontend): align frontend sso code with refactors

### PR DESCRIPTION
**Pull request by RyanHolstien**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: datahub-frontend/app/controllers/SsoCallbackController.java [hunk 1]
```diff
@@ -126,7 +133,18 @@ public Result perform(
   }
 
   private boolean shouldHandleCallback(final String protocol) {
-    return _ssoManager.isSsoEnabled()
-        && _ssoManager.getSsoProvider().protocol().getCommonName().equals(protocol);
+    if (!_ssoManager.isSsoEnabled()) {
+      return false;
+    }
+    updateConfig();
```
Comment (jjoyce0510) [inline datahub-frontend/app/controllers/SsoCallbackController.java:139]: My original impression was that this was doing a refresh of the SSO provider by fetching configs from the backend... Is that no longer required?

**Review by jjoyce0510 — APPROVED** [review]
Thank you for taking this on!

File: datahub-frontend/app/controllers/SsoCallbackController.java (hunk 1 above)
Comment (RyanHolstien) [reply datahub-frontend/app/controllers/SsoCallbackController.java:139]: That happens in a different part, see SSOManager.getDynamicSettings. That makes a call out to the AuthServiceController getting the settings from GMS.

Comment (RyanHolstien) [reply datahub-frontend/app/controllers/SsoCallbackController.java:139]: Do you mean the OIDC backend or GMS? There is separate logic to pull the configuration from GMS in SSOManager.getDynamicSettings which calls out to the AuthServiceController and the OIDC backend update happens in a separate flow controlled by AuthenticationController

**Pull request merged**